plugins/woocommerce/includes/data-stores/class-wc-product-variable-data-store-cpt.php

```diff
@@ -274,6 +285,13 @@ public function read_price_data( &$product, $for_display = false ) {
 		if ( empty( $this->prices_array[ $price_hash ] ) ) {
 			$transient_cached_prices_array = array_filter( (array) json_decode( strval( get_transient( $transient_name ) ), true ) );
 
+			// If the prices are not valid, reset the transient cache.
+			if ( ! $this->validate_prices_data( $transient_cached_prices_array, $transient_version ) ) {
+				$transient_cached_prices_array = array(
+					'version' => $transient_version,
+				);
+			}
+
 			// If the product version has changed since the transient was last saved, reset the transient cache.
 			if ( ! isset( $transient_cached_prices_array['version'] ) || $transient_version !== $transient_cached_prices_array['version'] ) {
```

Comment:
Are this check and the one in [L785](https://github.com/woocommerce/woocommerce/pull/53603/files#diff-c4ec3aeace51d4a1a8c8c49cc02b4046f9c650912cf7468426854d9595ac1755R785) checking the same? Wondering if we could get rid of one of them.

Reply:
Your right, done in https://github.com/woocommerce/woocommerce/pull/53603/commits/cbfbc24d19bd546e8e894706974554869421f308